### PR TITLE
Don't access pad_token_id if there is no pad_token

### LINKED ISSUE #4764: GPT2TokenizerFast raises pad_token error even if not used
# 🐛 Bug

## Information

Model: GPT2

Language: English

Encoding with the `GPT2TokenizerFast` causes a `pad_token` error to be sent to stderr, despite not attempting to access that property.

## To reproduce

```import transformers
from transformers import GPT2TokenizerFast

tokenizer = GPT2TokenizerFast.from_pretrained('gpt2')

res = tokenizer.encode("This is a sentence")

print(transformers.__version__)
```

Output:
```
Using pad_token, but it is not set yet.
2.11.0
```

## Expected behavior

I'm aware that GPT-2 doesn't include a pad token (#2630) - I haven't tried to use it. I would expect no error to be displayed until I try to access that property.

## Environment info

- `transformers` version: 2.11.0
- Platform: Ubuntu
- Python version: 3.6.9
- PyTorch version (GPU?): 1.5.0 with CUDA
- Tensorflow version (GPU?): n/a
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### PR DESCRIPTION
**Pull request by sgugger**
When using the `encode`  method of a fast tokenizer, we end up here and accessing `pad_token_id` may log an error even when no padding is done. This PR corrects that.

This fixes #4764 .

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
LGTM, also pinging @mfuntowicz for interface of Rust tokenizer and transformers tokenizer

**Review by patrickvonplaten — APPROVED** [review]

**Review by LysandreJik — APPROVED** [review]
LGTM
